Assets/Scripts/ExplotionController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Unity.Mathematics;
+using Unity.VisualScripting;
+using UnityEngine;
+using UnityEngine.VFX;
+
+public class ExplotionController : MonoBehaviour
+{
+    [SerializeField] 
+    public int damage;
+
+    [SerializeField]
+    private VisualEffect visualEffect;
+
+    [SerializeField]
+    public float radius;
+
+    [SerializeField]
+    private LayerMask hitBoxLayers;
+    
+    private HashSet<HealthController> hitHealthControllers = new HashSet<HealthController>();
+    
+
+    private void Start()
+    {
+        visualEffect.enabled = false;
+    }
+
+    public void OnDrawGizmos()
+    {
+        Gizmos.DrawWireSphere(transform.position,radius);
+    }
+
+    public void Explode()
+    {
+        visualEffect.enabled = true;
+        visualEffect.SendEvent("OnPlay");
+        Collider[] colliderList = Physics.OverlapSphere(transform.position, radius, hitBoxLayers);
+        
+        foreach (Collider collider in colliderList)
+        {
+            HitOrMiss(collider);
+        }
+    }
+
+    private void HitOrMiss(Collider collider)
+    {
+        HitboxController controller = collider.GetComponent<HitboxController>();
+        if (!controller.health || !hitHealthControllers.Contains(controller.health))
```

Comment:
Please check that the controller itself is not null (`!controller`) as well, as I suspect that's what colliders without a HitboxController will return in production mode.

Reply:
Never mind, turned out to not be necessary since we're limited by layer!